cloudbridge/cloud/providers/gce/resources.py

```diff
@@ -1063,3 +1068,253 @@ def delete(self):
                                          address=self._ip['name'])
                                  .execute())
        self._provider.wait_for_operation(response, region=self._region)
+
+
+class GCEVolume(BaseVolume):
+
+    VOLUME_STATE_MAP = {
+        'RESTORING': VolumeState.CONFIGURING,
+        'PENDING': VolumeState.CONFIGURING,
+        'READY': VolumeState.AVAILABLE,
+        'DONE': VolumeState.AVAILABLE,
+        'RUNNING': VolumeState.IN_USE,
+    }
+
+    def __init__(self, provider, volume):
+        super(GCEVolume, self).__init__(provider)
+        self._volume = volume
+
+    @property
+    def id(self):
+        return self._volume.get('selfLink')
+
+    @property
+    def name(self):
+        """
+        Get the volume name.
+        """
+        return self._volume.get('name')
+
+    @name.setter
+    # pylint:disable=arguments-differ
+    def name(self, value):
+        """
+        Set the volume name.
+        """
+        raise NotImplementedError('Not supported by this provider.')
+
+    @property
+    def description(self):
+        return self._volume.get('description')
+
+    @description.setter
+    def description(self, value):
+        raise NotImplementedError('Not supported by this provider.')
+
+    @property
+    def size(self):
+        return self._volume.get('sizeGb')
+
+    @property
+    def create_time(self):
+        return self._volume.get('creationTimestamp')
+
+    @property
+    def zone_id(self):
+        return self._volume.get('zone')
+
+    @property
+    def source(self):
+        return self._volume.get('sourceSnapshot')
+
+    @property
+    def attachments(self):
+        if 'users' in self._volume and len(self._volume) > 0:
+            return BaseAttachmentInfo(self,
+                                      self._volume.get('users')[0],
+                                      None)
+        else:
+            return None
+
+    def attach(self, instance, device):
```

Comment:
@baizhang @mbookman The device here means the blk device id, and not the mount point name. I think that the attach method will need to use or return a device id as otherwise, there would be no way to subsequently mount a disk without knowing the device id. This post suggests that it's the device_name parameter: https://stackoverflow.com/questions/33210664/programmatically-createmount-disk-from-within-google-compute-vm